api_modules/sessionize.py

```diff
@@ -1,5 +1,122 @@
+            soup = BeautifulSoup(response.content, 'html.parser')
+
+            if type == "Speakers":
+                rows.append(['id', 'name', 'tagline', 'bio', 'photo']) 
+                top_element = soup.find('ul', class_="sz-speakers--list")
+                if not top_element:
+                    raise RuntimeError('Unexpected server response!')
+                for row in top_element.find_all('li', class_='sz-speaker'):
+                    id = row.get('data-speakerid', '')
+                    name = text_or_default(row.find('h3', class_='sz-speaker__name'), "")
+                    tagline = text_or_default(row.find('h4', class_='sz-speaker__tagline'), "")
+                    bio = text_or_default(row.find('p', class_='sz-speaker__bio'), "")
+                    bio = bio.replace('<br>', '\n')
+                    photo = ''
+                    photo_el = row.find('div', class_='sz-speaker__photo')
+                    if photo_el:
+                        photo_img_el = photo_el.find('img')
+                        if photo_img_el:
+                            photo = photo_img_el.get('src', '')
+                    rows.append([id, name, tagline, bio, photo]) 
+            
+            elif type == "Sessions":
+                rows.append(['id', 'speaker_id', 'speaker', 'title', 'room_id', 'room', 'start_at', 'end_at', 'description']) 
+                top_element = soup.find('ul', class_="sz-sessions--list")
+                if not top_element:
+                    raise RuntimeError('Unexpected server response!')
+                for row in top_element.find_all('li', class_='sz-session'):
+                    id = row.get('data-sessionid', '')
+                    title = text_or_default(row.find('h3', class_='sz-session__title'), "")
+                    description = text_or_default(row.find('p', class_='sz-session__description'), "")
+                    description = description.replace('<br>', '\n')
+                    room, room_id = '', ''
+                    room_el = row.find('div', class_='sz-session__room')
+                    if room_el:
+                        room = room_el.text
+                        room_id = room_el.get('data-roomid', '')
+                    time_el = row.find('div', class_='sz-session__time')
+                    start_at, end_at = '', ''
+                    if time_el:
+                        time_str = time_el.get('data-sztz', '')
+                        if time_str:
+                            time_arr = time_str.split('|')
+                            if len(time_arr) >= 3:
+                                start_at = time_arr[2]
+                            if len(time_arr) >= 4:
+                                end_at = time_arr[3]
+                    speaker_id, speaker = '', ''
+                    speakers_el = row.find('ul', class_='sz-session__speakers')
+                    if speakers_el:
+                        for speaker_el in speakers_el.find_all('li'):
```

Comment:
Putting multiple speakers in a single comma-separated field might make it more complicated to import the data somewhere else. An alternative that is more common in CSV files is to add `speaker1`, `speaker2`, `speaker3`, etc. fields (as many as we think are needed) and then have the speaker entered in separate columns.

Maybe this is okay for now, and you can then change the format if needed.

Reply:
I am going to improve this also.